common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -257,4 +257,31 @@ public Properties cryptoConf() {
     return CryptoUtils.toCryptoConf("spark.network.crypto.config.", conf.getAll());
   }
 
+  /**
+   * When memory usage of Netty is above this water mark, it's regarded as memory shortage.
+   */
+  public Long nettyMemWaterMark() {
+    return JavaUtils.byteStringAsBytes(conf.get("spark.network.netty.memCostWaterMark", "1g"));
```

Comment:
This is only for shuffle service. Right? The config should start with `spark.shuffle`. I'm okey with one config since Netty either uses heap memory or off-heap memory.